core/services/pipeline/models.go

```diff
@@ -50,8 +50,9 @@ type Run struct {
 	PipelineTaskRuns []TaskRun        `json:"taskRuns"`
 	State            RunStatus        `json:"state"`
 
-	Pending   bool
-	FailEarly bool
+	Pending bool
+	// FailEarlySilently implies an error has happened and a task has been marked to failEarly
+	FailEarlySilently bool
```

Comment:
wondering if we should just change `failEarly` to `failSilently` - might be less confusing?

Reply:
in the task? idk cause it does 'fail early' as it does mark the other tasks canceled. and nothing in the task is done silently only once it hits the run level then things are done silently. 

actually I am going to change FailEarlySilently to just FailSilently because once it hits the run level its actually not doing anything early just silently.

So the task fails early and when that bubbles to the run it fails silently. I think hopefully that makes more sense.

Reply:
I'm pretty confused by fail early and fail silently semantics. Could you add more descriptive comment here about how it affects further pipeline tasks

Reply:
I tried to improve the comments. failEarly is a flag on a ethCall task. if an ethcall task with the failEarly flag fails during a pipeline run then it cancels the rest of the jobs and bubbles up to the run where the failSilently flag is flipped so the run knows to cleaning things up and not store the failed run in the db.